### DISCUSSION TITLE
Baseline file overriding the previous issues

### DISCUSSION THREAD

**Discussion by NitinPraksash9911** [detekt/detekt#4710] [Q&A] [answered]
Hey Team,

I have one query, after running this command `./gradlew detektBaseline` my base line file overriding the previous issues, so is there any way instead of overriding the baseline's previous issues, we can add more issues in the baseline file...

Please help me in this because I am facing lots of issues due to this behaviour

**Comment by BraisGabin** [comment, marked as answer]
I think that this is a question suffer from the xyproblem https://xyproblem.info/. It's strage that you need what you are asking for. What do you want to do? Why do you need this?

Anywat: a baseline have "two roots" `CurrentIssues` and `ManuallySuppressedIssues`. detekt updates the `CurrentIssues` automatically because if you fix them there are no reason to keep them. If you want to suppress some issues "manually" and keep them you should add them to `ManuallySuppressedIssues`.

**Reply by NitinPraksash9911** [reply]
@BraisGabin but what's about the issues present in the `baseline.xml`, and my question is i want add more issues in my `baseline.xml` instead of overriding the previous one, how can i do this?

**Reply by BraisGabin** [reply]
> If you want to suppress some issues "manually" and keep them you should add them to ManuallySuppressedIssues.

Move the ones that you want to keep to ManuallySuppressedIssues.

But again, this is strange. What's is your use case? Why do you need this?

**Reply by NitinPraksash9911** [reply]
My use-case is some-time we are in hurry to push the code where we don't have the to solve the all issues of detekt, by adding that issues in `baseline.xml,` we can pick them sometime in the future

**Reply by BraisGabin** [reply]
And why is detekt removing old issues? I mean, if it remove those issues is that they don't exist anymore so it should be ok to remove them, right?

In the company I work we have a daily task that regenerates all the baselines and commit them. Because we don't allow to add any new issue what this task does is to remove the issues that were fixed that day so they can't appear again.

And, sometimes, we also add some new issues to the baselines because we are in hurry but if that task remove old issues is not a problem, that means that the issue is no longer there. But this usually doesn't happen because our daily task keep our baselines up to date.

**Reply by NitinPraksash9911** [reply]
But in my case old issue which are still not resolved are overriding after running the gradle baseline command, so i just want to update the baseline.xml with new issue

**Reply by BraisGabin** [reply]
That's odd. It sounds like an issue or a bad configuration on your end.

**Reply by NitinPraksash9911** [reply]
Can you check what is wrong with my configuration?

```
apply {
    plugin("io.gitlab.arturbosch.detekt")
    plugin("org.jlleitschuh.gradle.ktlint")
}

ktlint {
    debug.set(false)
    verbose.set(true)
    android.set(false)
    outputToConsole.set(true)
    ignoreFailures.set(true)
//    disabledRules = ["no-wildcard-imports"]
    baseline = file("$rootDir/config/ktlint/ktlint-baseline.xml")
    filter {
        exclude("**/generated/**")
        include("**/kotlin/**")
    }
}

detekt {
    buildUponDefaultConfig = true // preconfigure defaults
    allRules = false // activate all available (even unstable) rules.
    config = files("$rootDir/config/detekt/detekt.yml")
    // point to your custom config defining rules to run, overwriting default behavior
    //warnings listed in the baseline file will be ignored! You can update the baseline file at any time by re-running ./gradlew detektBaseline.
    baseline = file("$rootDir/config/detekt/baseline.xml")
    reports {
        //you can find the generated reports in app/build/reports/detekt/detekt.html
        txt.enabled = false
        html.enabled = true
        xml.enabled = false
    }
}
```

**Reply by BraisGabin** [reply]
It seems that your issue is here: `file("$rootDir/config/detekt/baseline.xml")`. You are storing the baseline of ALL your modules in the same file. detekt gradle plugin is made to work at a module level. Set `file("baseline.xml")` and it will work. More information about this here: #1589

**Reply by NitinPraksash9911** [reply]
@BraisGabin Thanks man, setting this `file("baseline.xml")` has solved my issue
